homeassistant/components/sensor/sma.py

```diff
@@ -107,18 +113,24 @@ def _check_sensor_schema(conf):
     hass.bus.async_listen(EVENT_HOMEASSISTANT_STOP, async_close_session)
 
     backoff = 0
+    backoff_step = 0
 
     async def async_sma(event):
         """Update all the SMA sensors."""
-        nonlocal backoff
+        nonlocal backoff, backoff_step
         if backoff > 1:
             backoff -= 1
             return
 
         values = await sma.read(used_sensors)
-        if values is None:
-            backoff = 10
+        if not values:
+            try:
+                backoff = [1, 1, 1, 6, 6, 6, 30, 30][backoff_step]
```

Comment:
This looks ok. I would probably instead use a start value that is multiplied and cap it with min and max values. The value would represent the time until next try. Eg:
https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/netgear_lte.py#L110

But maybe we need a different approach here? You know the integration best.

Reply:
Let’s keep this. I will change to 1,1,1,6,30 and end with the 60

Reasoning
The current implementation ensures multiples of SCAN_INTERVAL, which feels consistent for reporting. Can likely do time based delay, but it will need to take S_I into account.

The library is fairly resilient and that why I have the many 1xS_I to start. If something goes wrong (a) might be network based or (b) not sure about the exact timeout when the user will be available(might have logged in through browser etc)